### PR TITLE
Move logic from GetHabboGroupBadgesEvent

### PR DESCRIPTION
**Pull request by Shetlands**

### CODE REVIEW

File: HabboHotel/Groups/GroupManager.cs [hunk 1]
```diff
@@ -169,4 +171,34 @@ public List<Group> GetGroupsForUser(int userId)
         }
         return groups;
     }
+
+    public Dictionary<int, string>? GetHabboGroupBadges(Habbo habbo)
```
Comment (80O) [inline HabboHotel/Groups/GroupManager.cs:175]: Method name doesnt reflect what its actually doing. This doesnt get the GroupBadges for a given Habbo.
This gets the badges for all users in a room.
The Habbo isn't needed as argument. Pass the room instead.

File: HabboHotel/Groups/GroupManager.cs [hunk 2]
```diff
@@ -169,4 +171,34 @@ public List<Group> GetGroupsForUser(int userId)
...
+        var badges = new Dictionary<int, string>();
+        foreach (var user in room.GetRoomUserManager().GetRoomUsers().ToList())
+        {
+            if (user.IsBot || user.IsPet || user.GetClient() == null || user.GetClient().GetHabbo() == null)
+                continue;
+            if (user.GetClient().GetHabbo().GetStats().FavouriteGroupId == 0 || badges.ContainsKey(user.GetClient().GetHabbo().GetStats().FavouriteGroupId))
+                continue;
+            if (!TryGetGroup(user.GetClient().GetHabbo().GetStats().FavouriteGroupId, out var group))
+                continue;
+            if (!badges.ContainsKey(group.Id))
```
Comment (80O) [inline HabboHotel/Groups/GroupManager.cs:191]: If statement is obsolete. Already checked on line 187.

File: HabboHotel/Groups/GroupManager.cs [hunk 3]
```diff
@@ -169,4 +171,34 @@ public List<Group> GetGroupsForUser(int userId)
...
+            return null;
+        var room = habbo.GetClient().GetHabbo().CurrentRoom;
+        if (room == null)
+            return null;
+        var badges = new Dictionary<int, string>();
+        foreach (var user in room.GetRoomUserManager().GetRoomUsers().ToList())
+        {
+            if (user.IsBot || user.IsPet || user.GetClient() == null || user.GetClient().GetHabbo() == null)
+                continue;
+            if (user.GetClient().GetHabbo().GetStats().FavouriteGroupId == 0 || badges.ContainsKey(user.GetClient().GetHabbo().GetStats().FavouriteGroupId))
```
Comment (80O) [inline HabboHotel/Groups/GroupManager.cs:187]: 5 times a call to the same field. Assign it to a local variable instead.

File: HabboHotel/Groups/GroupManager.cs [hunk 4]
```diff
@@ -169,4 +171,34 @@ public List<Group> GetGroupsForUser(int userId)
...
+                continue;
+            if (!TryGetGroup(user.GetClient().GetHabbo().GetStats().FavouriteGroupId, out var group))
+                continue;
+            if (!badges.ContainsKey(group.Id))
+                badges.Add(group.Id, group.Badge);
+        }
+        if (habbo.GetClient().GetHabbo().GetStats().FavouriteGroupId > 0)
```
Comment (80O) [inline HabboHotel/Groups/GroupManager.cs:194]: If statement is obsolete. habbo is already part of the list of users iterated on line 183

**Comment by Shetlands** [comment]
done

**Pull request merged**